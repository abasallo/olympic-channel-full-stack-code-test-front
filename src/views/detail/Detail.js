import React from 'react'

import Toolbar from '@material-ui/core/Toolbar'

import logo from '../../images/logo.png'

import {
  AppBar,
  Logo,
  Container,
  CopyrightBox,
  DetailDiv,
  HeaderDetailDiv,
  BioDetailDiv,
  StyledPaper,
  StyledTable
} from './Detail.styled.components'

import Copyright from '../../components/Copyright/Copyright'

import constants from '../../modules/constants'

import { useParams } from 'react-router'

import { Query } from 'react-apollo'

import { GET_ATHLETE } from '../../services/graphql/AthleteQueries'

import ReactMarkdown from 'react-markdown'
import TableContainer from '@material-ui/core/TableContainer'

import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'
import TableBody from '@material-ui/core/TableBody'
import AthleteDetail from '../../components/AthleteDetail/AthleteDetail'

const renderMedals = (athleteResults) => (
  <TableContainer component={StyledPaper}>
    <StyledTable>
      <TableHead>
        <TableRow>
          <TableCell align="center">Year</TableCell>
          <TableCell align="center">Game</TableCell>
          <TableCell align="center">Gold</TableCell>
          <TableCell align="center">Silver</TableCell>
          <TableCell align="center">Bronze</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {athleteResults.map((result) => (
          <TableRow key={result.id}>
            <TableCell align="center">{result.game.year}</TableCell>
            <TableCell align="center">{result.game.city}</TableCell>
            <TableCell align="center">{result.gold}</TableCell>
            <TableCell align="center">{result.silver}</TableCell>
            <TableCell align="center">{result.bronze}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </StyledTable>
  </TableContainer>
)

const renderAthlete = (athlete) => (
  <DetailDiv>
    <HeaderDetailDiv>
      <AthleteDetail athlete={athlete} />
    </HeaderDetailDiv>
    <div>
      <h1>Medals</h1>
      {renderMedals(athlete.athleteResults)}
    </div>
    <div>
      <h1>Bio</h1>
      <BioDetailDiv>
        <ReactMarkdown source={athlete.bio} />
      </BioDetailDiv>
    </div>
  </DetailDiv>
)

const Detail = () => {
  const { id } = useParams()
  return (
    <React.Fragment>
      <AppBar position="static">
        <Toolbar>
          <Logo src={logo} alt={constants.LOGO_ALT} />
        </Toolbar>
      </AppBar>
      <Container>
        <Query query={GET_ATHLETE} variables={{ id }}>
          {({ loading, error, data }) => {
            if (loading) return 'Loading...'
            if (error) return `Error! ${error.message}`
            return renderAthlete(data.getAthlete)
          }}
        </Query>
      </Container>
      <CopyrightBox mt={8}>
        <Copyright />
      </CopyrightBox>
    </React.Fragment>
  )
}

export default Detail
