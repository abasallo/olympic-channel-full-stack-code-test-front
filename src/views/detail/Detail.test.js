import React from 'react'

import Detail from './Detail'

test('Renders properly', () => expect(<Detail match={{ params: { id: 1 }, isExact: true, path: '', url: '' }} />).toMatchSnapshot())
