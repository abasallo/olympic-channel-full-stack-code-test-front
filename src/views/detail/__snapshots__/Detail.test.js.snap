// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<Detail
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": 1,
      },
      "path": "",
      "url": "",
    }
  }
/>
`;
